File: Python/Buyer.py
from Person import Person
from Market import Market

class Buyer(Person):
    def __init__(self, id_person=0, name="", address="", budget=0.0):
        super().__init__(id_person, name, address)
        self.max_budget = budget
        self.list_market = []

    # ===== Getter methods =====
    def get_max_budget(self):
        return self.max_budget

    def get_list_market(self):
        return self.list_market

    # ===== Setter methods =====
    def set_max_budget(self, budget):
        self.max_budget = budget

    def set_list_market(self, daftar):
        self.list_market = daftar

    # Tambah Market ke daftar
    def add_market(self, market: Market):
        self.list_market.append(market)

    # Display Buyer info
    def display_buyer(self, indent=0):
        sp = " " * indent
        print(f"{sp}Buyer ID   : {self.get_id_person()}")
        print(f"{sp}Name       : {self.get_name()}")
        print(f"{sp}Address    : {self.get_address()}")
        print(f"{sp}Max Budget : {self.get_max_budget()}\n")

        if self.list_market:
            print(f"{sp}    Markets:")
            for m in self.list_market:
                m.display_market(indent + 8)  # tampilkan Market beserta Seller/Farmer
        print()

    def __del__(self):
        pass

File: Java/Buyer.java
import java.util.ArrayList;
import java.util.List;

public class Buyer extends Person {
    private double maxBudget;
    private List<Market> listMarket;

    // Default constructor
    public Buyer() {
        super(); // panggil constructor Person()
        this.maxBudget = 0.0;
        this.listMarket = new ArrayList<>();
    }

    // Parameterized constructor
    public Buyer(int id, String name, String address, double budget) {
        super(id, name, address); // panggil constructor Person dengan parameter
        this.maxBudget = budget;
        this.listMarket = new ArrayList<>();
    }

    // ===== Getter methods =====
    public double getMaxBudget() {
        return maxBudget;
    }

    public List<Market> getListMarket() {
        return listMarket;
    }

    // ===== Setter methods =====
    public void setMaxBudget(double budget) {
        this.maxBudget = budget;
    }

    public void setListMarket(List<Market> daftar) {
        this.listMarket = daftar;
    }

    // Tambah Market ke daftar
    public void addMarket(Market market) {
        this.listMarket.add(market);
    }

    // Display Buyer detail
    public void displayBuyer(int indent) {
        String sp = " ".repeat(indent);
        System.out.println(sp + "Buyer ID   : " + getIdPerson());
        System.out.println(sp + "Name       : " + getName());
        System.out.println(sp + "Address    : " + getAddress());
        System.out.println(sp + "Max Budget : " + getMaxBudget());
        System.out.println();

        if (!listMarket.isEmpty()) {
            System.out.println(sp + "    Markets:");
            for (Market m : listMarket) {
                m.displayMarket(indent + 8); // tampilkan Market beserta Seller/Farmer
            }
        }
        System.out.println();
    }
}
